src/deq_eid/main.py

```diff
@@ -0,0 +1,363 @@
+
+    def _initialize_supervisor(self):
+        """A helper method to set up logging and supervisor
+
+        Returns:
+            Supervisor: The supervisor object used for sending messages
+        """
+
+        skid_logger = logging.getLogger(config.SKID_NAME)
+        skid_logger.setLevel(config.LOG_LEVEL)
+        palletjack_logger = logging.getLogger("palletjack")
+        palletjack_logger.setLevel(config.LOG_LEVEL)
+
+        cli_handler = logging.StreamHandler(sys.stdout)
+        cli_handler.setLevel(config.LOG_LEVEL)
+        formatter = logging.Formatter(
+            fmt="%(levelname)-7s %(asctime)s %(name)15s:%(lineno)5s %(message)s", datefmt="%Y-%m-%d %H:%M:%S"
+        )
+        cli_handler.setFormatter(formatter)
+
+        log_handler = logging.FileHandler(self.log_path, mode="w")
+        log_handler.setLevel(config.LOG_LEVEL)
+        log_handler.setFormatter(formatter)
+
+        skid_logger.addHandler(cli_handler)
+        skid_logger.addHandler(log_handler)
+        palletjack_logger.addHandler(cli_handler)
+        palletjack_logger.addHandler(log_handler)
+
+        #: Log any warnings at logging.WARNING
+        #: Put after everything else to prevent creating a duplicate, default formatter
+        #: (all log messages were duplicated if put at beginning)
+        logging.captureWarnings(True)
+
+        skid_logger.debug("Creating Supervisor object")
+        self.supervisor = Supervisor(handle_errors=False)
+        sendgrid_settings = config.SENDGRID_SETTINGS
+        sendgrid_settings["api_key"] = self.secrets.SENDGRID_API_KEY
+        self.supervisor.add_message_handler(
+            SendGridHandler(
+                sendgrid_settings=sendgrid_settings, client_name=config.SKID_NAME, client_version=version.__version__
+            )
+        )
+
+    def _remove_log_file_handlers(self):
+        """A helper function to remove the file handlers so the tempdir will close correctly"""
+
+        loggers = [logging.getLogger(config.SKID_NAME), logging.getLogger("palletjack")]
+
+        for logger in loggers:
+            for handler in logger.handlers:
+                try:
+                    if self.log_name in handler.stream.name:
+                        logger.removeHandler(handler)
+                        handler.close()
+                except Exception:
+                    pass
+
+    def _get_incidents(self) -> GeoAccessor:
+        self.skid_logger.info("loading environmental incident records from Salesforce...")
+        records = helpers.SalesForceRecords(
+            self.salesforce_extractor,
+            config.INCIDENTS_SF_API,
+            config.INCIDENTS_FIELDS,
+            where_clause="Utm_N_Y_7_dgts__c != null and Utm_E_X_6_dgts__c != null",
+        )
+        records.extract_data_from_salesforce()
+
+        self.skid_logger.info("converting to spatial dataframe...")
+        sdf = GeoAccessor.from_xy(records.df, "Easting", "Northing", sr=26912)
```

Comment:
question: I've usually seen this done as `sdf = pd.DataFrame.spatial.from_xy(...)`. My understanding is that GeoAccessor extends the pd.DataFrame namespace with .spatial so that you can use the spatial methods similar to other dataframe methods.

Reply:
I'm not sure what the question is other than why I did it this way? The answer to that is probably ¯\_(ツ)_/¯. Is there any reason to use one way over the other? The only one that I can think of is that you don't have to ignore the lint rule for an unused import.